File: python/982.py
from collections import Counter
from typing import List

class Solution:
    def countTriplets(self, nums: List[int]) -> int:
        m = Counter(a & b for a in nums for b in nums)
        return sum(n for x, n in m.items() for c in nums if x & c == 0)


s = Solution()
print(s.countTriplets([2,1,3]))

File: go/982.go
package main

func countTriplets(nums []int) (ans int) {
	cnt := [1 << 16]int{}
	for _, a := range nums {
		for _, b := range nums {
			cnt[a&b]++
		}
	}
	for x, n := range cnt {
		for _, c := range nums {
			if x&c == 0 {
				ans += n
			}
		}
	}
	return ans
}

// func main() {

// }
